video/out/vo_dmabuf_wayland.c

```diff
@@ -241,21 +244,33 @@ static int reconfig(struct vo *vo, struct mp_image_params *params)
     struct priv *p = vo->priv;
     struct vo_wayland_state *wl = vo->wl;
 
-    if (!p->solid_buffer_pool) {
-        int width = 1;
-        int height = 1;
-        int stride = MP_ALIGN_UP(width * 4, 16);
-        int fd = vo_wayland_allocate_memfd(vo, stride);
-        if (fd < 0)
-            return VO_ERROR;
-        p->solid_buffer_pool = wl_shm_create_pool(wl->shm, fd, height * stride);
-        if (!p->solid_buffer_pool)
-            return VO_ERROR;
-        p->solid_buffer = wl_shm_pool_create_buffer(p->solid_buffer_pool, 0, width, height, stride, WL_SHM_FORMAT_XRGB8888);
+    if (!p->solid_buffer) {
```

Comment:
Just carrying over comments from irc, but all of this stuff could all probably go in `preinit`.